x/axelarnet/message_handler.go

```diff
@@ -102,8 +102,9 @@ func OnRecvMessage(ctx sdk.Context, k keeper.Keeper, ibcK keeper.IBCKeeper, n ty
 		return channeltypes.NewErrorAcknowledgement(err)
 	}
 
-	// skip if packet not sent to Axelar message sender account
-	if data.GetReceiver() != types.AxelarGMPAccount.String() {
+	// Skip if packet not sent to Axelar message sender account.
+	// Receiver must be valid at this point. Otherwise, the transfer handler in upper layer would reject the packet.
```

Comment:
I'd prefer to not increase our reliance on the middleware ordering, but not sure if i like the alternatives here too, haha